services/web/server/src/simcore_service_webserver/login/_2fa_api.py

```diff
@@ -95,33 +99,46 @@ async def send_sms_code(
     twilio_messaging_sid: str,
     twilio_alpha_numeric_sender: str,
     first_name: str,
+    user_id: UserID | None = None,
 ):
-    create_kwargs = {
-        "messaging_service_sid": twilio_messaging_sid,
-        "to": phone_number,
-        "body": f"Dear {first_name}, your verification code is {code}",
-    }
-    if twilo_auth.is_alphanumeric_supported(phone_number):
-        create_kwargs["from_"] = twilio_alpha_numeric_sender
-
-    def _sender():
-        log.info(
-            "Sending sms code to %s from product %s",
-            f"{phone_number=}",
-            twilio_alpha_numeric_sender,
+    try:
+        create_kwargs = {
+            "messaging_service_sid": twilio_messaging_sid,
+            "to": phone_number,
+            "body": f"Dear {first_name}, your verification code is {code}",
+        }
+        if twilo_auth.is_alphanumeric_supported(phone_number):
+            create_kwargs["from_"] = twilio_alpha_numeric_sender
+
+        def _sender():
+            log.info(
+                "Sending sms code to %s from product %s",
+                f"{phone_number=}",
+                twilio_alpha_numeric_sender,
+            )
+            #
+            # SEE https://www.twilio.com/docs/sms/quickstart/python
+            #
+            client = Client(twilo_auth.TWILIO_ACCOUNT_SID, twilo_auth.TWILIO_AUTH_TOKEN)
+            message = client.messages.create(**create_kwargs)
+
+            log.debug(
+                "Got twilio client %s",
+                f"{message=}",
+            )
+
+        await asyncio.get_event_loop().run_in_executor(executor=None, func=_sender)
+
+    except TwilioException as exc:
+        error_code = create_error_code(exc)
+        more_extra: LogExtra = get_log_record_extra(user_id=user_id) or {}
```

Comment:
more_extra.. that is a funny name;) why not log_extra?

Reply:
Haha didn't notice, this is the existing code it was just moved here from `handlers_2fa`, I will rename.